src/bin/pg_dump/dump_babel_utils.c

```diff
@@ -541,3 +612,326 @@ updateExtConfigArray(Archive *fout, char ***extconfigarray, int nconfigitems)
+
+	if (!isBabelfishDatabase(fout) || bbf_db_name == NULL)
+		return;
+
+	if (tbinfo->dobj.namespace == NULL ||
+		strcmp(tbinfo->dobj.namespace->dobj.name, "sys") != 0 ||
+		!(strcmp(tbinfo->dobj.name, "babelfish_sysdatabases") == 0 ||
+		 strcmp(tbinfo->dobj.name, "babelfish_namespace_ext") == 0 ||
+		 strcmp(tbinfo->dobj.name, "babelfish_function_ext") == 0 ||
+		 strcmp(tbinfo->dobj.name, "babelfish_authid_user_ext") == 0 ||
+		 strcmp(tbinfo->dobj.name, "babelfish_view_def") == 0 ||
+		 strcmp(tbinfo->dobj.name, "babelfish_domain_mapping") == 0))
+		return;
+
+	is_builtin_db = (pg_strcasecmp(bbf_db_name, "master") == 0 ||
+			pg_strcasecmp(bbf_db_name, "tempdb") == 0 ||
+			pg_strcasecmp(bbf_db_name, "msdb") == 0)
+			? true : false;
+
+	resetPQExpBuffer(buf);
+	appendPQExpBufferStr(buf, "DECLARE _pg_dump_cursor CURSOR FOR SELECT ");
+	*nfields = 0;
+	for (i = 0; i < tbinfo->numatts; i++)
+	{
+		if (tbinfo->attisdropped[i])
+			continue;
+		/* Skip dbid column, we will generate new database id during restore. */
+		if (!is_builtin_db && strcmp(tbinfo->attnames[i], "dbid") == 0)
+			continue;
+		if (*nfields > 0)
+			appendPQExpBufferStr(buf, ", ");
+		appendPQExpBufferStr(buf, "a.");
+		appendPQExpBufferStr(buf, fmtId(tbinfo->attnames[i]));
+		(*nfields)++;
+	}
+
+	if (strcmp(tbinfo->dobj.name, "babelfish_sysdatabases") == 0)
+	{
+		appendPQExpBuffer(buf, " FROM ONLY %s a WHERE a.dbid = %d",
+						  fmtQualifiedDumpable(tbinfo), bbf_db_id);
+		/*
+		 * builtin db will already be present in the target server so
+		 * no need to dump catalog entry for it
+		 */
+		if (is_builtin_db)
+			appendPQExpBufferStr(buf, " LIMIT 0");
+	}
+	else if (strcmp(tbinfo->dobj.name, "babelfish_namespace_ext") == 0)
+	{
+		appendPQExpBuffer(buf, " FROM ONLY %s a WHERE a.dbid = %d",
+						  fmtQualifiedDumpable(tbinfo), bbf_db_id);
+		if (is_builtin_db)
+			appendPQExpBuffer(buf, " AND a.nspname NOT IN ('%s_dbo', '%s_guest')",
+							  escaped_bbf_db_name, escaped_bbf_db_name);
+	}
+	else if (strcmp(tbinfo->dobj.name, "babelfish_view_def") == 0)
+		appendPQExpBuffer(buf, " FROM ONLY %s a WHERE a.dbid = %d",
+						  fmtQualifiedDumpable(tbinfo), bbf_db_id);
+	else if (strcmp(tbinfo->dobj.name, "babelfish_function_ext") == 0)
+		appendPQExpBuffer(buf, " FROM ONLY %s a "
+						  "INNER JOIN sys.babelfish_namespace_ext b "
+						  "ON a.nspname = b.nspname "
+						  "WHERE b.dbid = %d",
+						  fmtQualifiedDumpable(tbinfo), bbf_db_id);
+	else if(strcmp(tbinfo->dobj.name, "babelfish_authid_user_ext") == 0)
+	{
+		appendPQExpBuffer(buf, " FROM ONLY %s a "
+						  "WHERE a.rolname IN ('dbo', 'db_owner', '%s_dbo', '%s_db_owner', '%s_guest') ",
+						  fmtQualifiedDumpable(tbinfo), escaped_bbf_db_name, escaped_bbf_db_name,
+						  escaped_bbf_db_name);
+		/*
+		 * Builtin db users will already be present in the target
+		 * server so no need to dump their catalog data.
+		 */
+		if (is_builtin_db)
+			appendPQExpBufferStr(buf, "LIMIT 0 ");
+	}
+	else
```

Comment:
Seems like it is only for `babelfish_domain_mapping`. Let's make it explicit and error out for the `else` block.

Reply:
Done!